addons/mass_mailing/models/mailing.py

```diff
@@ -1306,36 +1315,117 @@ def _unsubscribe_token(self, res_id, email):
         token = (self.env.cr.dbname, self.id, int(res_id), tools.ustr(email))
         return hmac.new(secret.encode('utf-8'), repr(token).encode('utf-8'), hashlib.sha512).hexdigest()
 
-    def _convert_inline_images_to_urls(self, body_html):
-        """
-        Find inline base64 encoded images, make an attachement out of
-        them and replace the inline image with an url to the attachement.
-        """
 
-        def _image_to_url(b64image: bytes):
-            """Store an image in an attachement and returns an url"""
-            attachment = self.env['ir.attachment'].create({
-                'datas': b64image,
-                'name': "cropped_image_mailing_{}".format(self.id),
-                'type': 'binary',})
+    def _image_to_url(self, b64image: bytes):
```

Comment:
Would be better to have a batch-enabled method, in order to avoid sequential code creating attachments in a row. Accumulate images to create and then create them in batch :) .

Reply:
Done here: https://github.com/odoo/odoo/pull/119325/commits/b51d2854350dbc3094f6491b58bbf34a00986a39